lib/internal/test_runner/test.js

```diff
@@ -172,9 +172,9 @@ class Test extends AsyncResource {
 
   constructor(options) {
     super('Test');
-
-    let { fn, name, parent, skip } = options;
+    let { fn, name, parent, skip, testNamePatterns } = options;
     const { concurrency, only, timeout, todo, signal } = options;
+    if (testNamePatternsfromCLI) testNamePatterns = testNamePatternsfromCLI
```

Comment:
i think a merge is better, imagine this scenario:

you are in a complex project and have some files that use "run()", you already have some filters there for some reason, do you want to add a global filter or just complement your filter for that special case, why change the files instead of just use the CLI?

i think it could increase the nodejs DX

Reply:
or a simple scenario, you are in the server or just in a terminal interface, and you want to debug 1 test only, just add the cli filter instead update the file.

Reply:
I think this will always be subjective, but IMO merging the two is the least good option. Inevitably there will be someone that doesn't want the merge behavior. I also think that `run()` and the CLI option are use cases that shouldn't overlap a ton in user code - `run()` is supposed to be a programmatic way of implementing things the CLI runner does.

Reply:
I think it is ok that when using `run` cli flags will be ignored